db/migrate/20210602210437_add_auth_invitation_to_project.rb

```diff
@@ -0,0 +1,5 @@
+class AddAuthInvitationToProject < ActiveRecord::Migration
+  def change
+    add_column :projects, :authentication_invitation, :string
```

Comment:
i'd say we should change the column type to text vs varchar (string) https://stackoverflow.com/questions/3354330/difference-between-string-and-text-in-rails
```suggestion
    add_column :projects, :authentication_invitation, :text
```

Reply:
Also it's best to set a default on this column to ensure we don't serialize different value types, i.e. `null` or `'possibly empty string or string with data in it'`

Read up on how to do that here https://github.com/ankane/strong_migrations#good-1

Gist is you add the column without a default, set the default and then we can backfill the column values manually. Backfilling can be done in a migration, a rake task or manually on the console. 

We tend to prefer running the data migration / backfill operations via rake tasks so we can test the code, make it resumeable and de-couple from the actual DDL migration (i.e. still migrate the schema even if the data can't be changed), e.g. https://github.com/zooniverse/panoptes/blob/e9001518f410907bd8e87627cd55f659882c4e6e/lib/tasks/migrate.rake#L78-L86

Reply:
I think I see what you mean here. so if i backfill this field, we don't want empty string (`''`)? Should I set it to 'default'?